consumer.go

```diff
@@ -75,8 +77,10 @@ func parsePosition(bytes []byte) (position, error) {
 }
 
 type segmentConsumer struct {
-	reader      *kafka.Reader
-	lastMsgRead *kafka.Message
+	reader *kafka.Reader
+	// unackMessages represents all messages which have been read but not acknowledged.
+	// They are ordered in the way they were read.
+	unackMessages []*kafka.Message
```

Comment:
It might make sense to limit the capacity of this slice, to prevent memory leaks when, for example, a message is "stuck".